action/src/utils.ts

```diff
@@ -167,4 +174,92 @@ export class Utils {
     public static isWindows() {
         return platform().startsWith('win');
     }
+    public static async validatePlatformUrl(jfrogUrl: string): Promise<void> {
+        const jfrogUrlFailure: string = 'JF_URL must point on your full platform URL, for example: https://mycompany.jfrog.io/, make sure the platform is up and running and accessible.'
+        //verify that the provided JFrog URL is valid and responsive
+        const pingUrl: string = jfrogUrl!.replace(/\/$/, '') + '/artifactory/api/system/ping';
+        const httpClient: HttpClient = new HttpClient();
+        let response: HttpClientResponse
+        try {
+             response = await httpClient.get(pingUrl);
+        }catch (error: any) {
+            throw new Error(jfrogUrlFailure + ', Error returned is ' + error.message);
+        }
+        if (response.message.statusCode == 200) {
+            const body: string = await response.readBody();
+            if (body == 'OK') {
+                return ;
+            }else
+                throw new Error(jfrogUrlFailure);
+        } else {
+           throw new Error(jfrogUrlFailure);
+        }
+    }
+
+    public static async getJfrogOIDCCredentials(jfrogUrl: string): Promise<void> {
+
+        const oidcProviderName: string = core.getInput(Utils.OIDC_INTEGRATION_PROVIDER_NAME);
+        if (!oidcProviderName) {
+            // no token is set if no oidc provider was configured
+            return ;
+        }
+        core.debug('Obtaining an access token through OpenID Connect...');
+        const audience: string = core.getInput(Utils.OIDC_AUDIENCE_ARG);
+        let jsonWebToken: string | undefined;
+        try {
+            core.debug('Fetching JSON web token');
+            jsonWebToken = await core.getIDToken(audience);
+        } catch (error: any) {
+            throw new Error(`Getting openID Connect JSON web token failed: ${error.message}`);
+        }
+
+        try {
+            return await this.initJfrogAccessTokenThroughOidcProtocol(jfrogUrl, jsonWebToken, oidcProviderName);
+        } catch (error: any) {
+            throw new Error(`OIDC authentication against JFrog platform failed, please check OIDC settings and mappings on the JFrog platform: ${error.message}`);
+        }
+    }
+
+
+    private static async initJfrogAccessTokenThroughOidcProtocol(
+        jfrogUrl: string,
+        jsonWebToken: string,
+        oidcProviderName: string,
+    ): Promise<void>     {
+        // assuming in this method that add parameters were provided
```

Comment:
```suggestion
        // Assuming in this method that add parameters were provided
```
Also, what is "add parameters"?